### PR TITLE
Preserve fill color on toggle enabled

### PR DESCRIPTION
**Pull request by volfied**
Addresses #2076 and #2077, casualties of a refactor.

### CODE REVIEW

File: src/js/jsx/sections/style/Fill.jsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ define(function (require, exports, module) {
             this.getFlux().actions.shapes.setFillEnabled(
                 this.props.document,
                 this.state.layers,
-                this.state.fill && this.state.fill.color || Color.DEFAULT,
+                this.state.fill && this.state.fill.colors.first() || Color.DEFAULT,
```
Comment (iwehrman) [inline src/js/jsx/sections/style/Fill.jsx]: Should this be the result of `collection.uniformValue`?

**Comment by ktaki** [comment]
#2076 is fixed, but I still see #2077.

**Comment by ktaki** [comment]
OK. I spoke too fast. I think #2080 contributes #2076.
Create a rect. Set fill color. Toggle the visibility. This works. Now, set the alpha to 90%. Toggle the visibility. It resets the fill color to black.

**Comment by volfied** [comment]
Now it also fixes #2080

**Comment by ktaki** [comment]
Confirmed the fixes for #2076 and #2080.

File: src/js/models/layerstructure.js [hunk 2]
```diff
@@ -1213,13 +1212,19 @@ define(function (require, exports, module) {
...
-                return map.set(layerID, Immutable.Map({
-                    fill: nextFill
-                }));
-            }, new Map()));
+        var nextLayers = Immutable.Map(layerIDs.reduce(function (map, layerID) {
+            var layer = this.byID(layerID),
+                fill = layer.fill;
+
+            if (!fill) {
+                throw new Error("Unable to set fill properties: no fill for layer " + layer.name);
```
Comment (iwehrman) [inline src/js/models/layerstructure.js]: Seems like it would be better to just make a new Fill here, but I guess that can wait for a later day.

**Comment by iwehrman** [comment]
Let's do this.

**Pull request merged**